tests/core/kernel-base-gadget-pair-single-reboot-failover/task.yaml

```
@@ -17,82 +21,115 @@ environment:
     if [ "$SNAP_ONE" = "core" ]; then
         SNAP_ONE=$core_snap
     elif [ "$SNAP_TWO" = "core" ]; then
         SNAP_TWO=$core_snap
     fi
 
+    # shellcheck source=tests/lib/prepare.sh
+    . "$TESTSLIB/prepare.sh"
+
+    inject_failure_configure_hook() {
+        if gojq --yaml-input --yaml-output '. | has("hooks")' "$1/meta/snap.yaml" | MATCH "true"; then
+            # shellcheck disable=SC2094
+            gojq --yaml-input --yaml-output '.hooks += {"configure"}' "$1/meta/snap.yaml" >> "$1/meta/snap.yaml"
+        else
+            # shellcheck disable=SC2094
+            gojq --yaml-input --yaml-output '. += {"hooks": {"configure"}}' "$1/meta/snap.yaml" >> "$1/meta/snap.yaml"
+            mkdir -p "$1/meta/hooks"
+        fi
+        cat <<EOF >"$1/meta/hooks/configure"
+    #!/bin/bash
+    # invalid hook
+    exit 1
+    EOF
+        chmod +x "$1/meta/hooks/configure"
+    }
+
     if [ "$SPREAD_REBOOT" = 0 ]; then
-        snap ack "$TESTSLIB/assertions/testrootorg-store.account-key"
-        "$TESTSTOOLS"/store-state setup-fake-store "$BLOB_DIR"
+        echo Downloading snaps needed
+        snap download pc-kernel --channel="$VERSION_ID"/edge
+
+        snap download "$core_snap" --edge
+        printf '\0' >> "$core_snap"_*.snap
 
         # break the pc-kernel snap if that is the last snap
         if [[ "$SNAP_TWO" == "pc-kernel" ]]; then
-            if os.query is-core18; then
-                unsquashfs -d pc-kernel-snap /var/lib/snapd/snaps/pc-kernel_*.snap
-                truncate -s 0 pc-kernel-snap/initrd.img
-            elif os.query is-core20; then
-                uc20_build_initramfs_kernel_snap pc-kernel.snap "$PWD" --inject-kernel-panic-in-initramfs
-            else
-                echo "unsupported Ubuntu Core system"
-                exit 1
-            fi
-            "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" pc-kernel --snap-blob "$PWD/pc-kernel-snap"
-        elif [[ "$SNAP_ONE" == "pc-kernel" ]]; then
-            "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" pc-kernel
+            unsquashfs -d pc-kernel-snap pc-kernel_*.snap
+            inject_failure_configure_hook ./pc-kernel-snap
+            rm pc-kernel_*.snap pc-kernel_*.assert
+            snap pack pc-kernel-snap
+            rm -rf pc-kernel-snap
         fi
 
-        # break the gadget snap if that is the second snap
-        if [[ "$SNAP_TWO" == "pc" ]]; then
-            unsquashfs -d pc-snap /var/lib/snapd/snaps/pc_*.snap
-            echo 'exit 1' >> ./pc-snap/snap/hooks/configure
-            "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" pc --snap-blob "$PWD/pc-snap"
-        elif [[ "$SNAP_ONE" == "pc" ]]; then
-            "$TESTSTOOLS"/store-state init-fake-refreshes "$BLOB_DIR" pc
-        fi
+        # prepare gadget update if gadget variant
+        # and break pc gadget if that is the last snap
+        if [[ "$SNAP_ONE" == "pc" ]] || [[ "$SNAP_TWO" == "pc" ]]; then
```

Nitpick, can we use the simpler comparison?
```suggestion
        if [ "$SNAP_ONE" = pc ] || [ "$SNAP_TWO" = pc ]; then
```